libs/cli/langchain_cli/namespaces/app.py

```diff
@@ -55,21 +54,55 @@ def new(
             is_flag=True,
         ),
     ] = None,
+    nointeractive: Annotated[
+        bool,
+        typer.Option(
+            "--nointeractive",
+            help="Don't prompt for any input",
+            is_flag=True,
+        ),
+    ] = False,
 ):
     """
     Create a new LangServe application.
     """
     has_packages = package is not None and len(package) > 0
-    pip_bool = False
-    if pip is None and has_packages:
-        pip_bool = typer.confirm(
-            "Would you like to `pip install -e` the template(s)?",
-            default=False,
+
+    if nointeractive:
+        if name is None:
+            raise typer.BadParameter("name is required when --nointeractive is set")
+        name_str = name
+        pip_bool = bool(pip)
+    else:
+        name_str = (
+            name if name else typer.prompt("What folder would you like to create?")
         )
+        if not has_packages:
+            package = []
+            package_prompt = "What package would you like to add? (leave blank to skip)"
+            while True:
+                package_str = typer.prompt(
+                    package_prompt, default="", show_default=False
+                )
+                if not package_str:
+                    break
+                package.append(package_str)
+                package_prompt = (
+                    f"{len(package)} added. Any more packages (leave blank to end)?"
+                )
+
+            has_packages = len(package) > 0
+
+        pip_bool = False
+        if pip is None and has_packages:
+            pip_bool = typer.confirm(
```

Comment:
I bet that a lot of users will not know what this means. Could we include an explanation about which default we recommend?

Reply:
Could also construct the exact command before this, and ask
```
To install these {n} templates with `pip`, run:

pip install -e ...

Would you like to run it? [y/N]
```
Which defaults to no `[y/N]`

Reply:
This is a bigger refactor. Will just rephrase as "Would you like to install these templates in your environment with pip?"